src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/CheckStayOrderDeadlineEndTaskHandler.java

```diff
@@ -0,0 +1,81 @@
+package uk.gov.hmcts.reform.civil.handler.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.camunda.bpm.client.task.ExternalTask;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.enums.YesOrNo;
+import uk.gov.hmcts.reform.civil.helpers.CaseDetailsConverter;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.genapplication.GAJudicialMakeAnOrder;
+import uk.gov.hmcts.reform.civil.service.CoreCaseDataService;
+import uk.gov.hmcts.reform.civil.service.search.OrderMadeSearchService;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.time.LocalDate.now;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.END_SCHEDULER_CHECK_STAY_ORDER_DEADLINE;
+import static uk.gov.hmcts.reform.civil.enums.dq.GeneralApplicationTypes.STAY_THE_CLAIM;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@ConditionalOnExpression("${response.deadline.check.event.emitter.enabled:true}")
+public class CheckStayOrderDeadlineEndTaskHandler implements BaseExternalTaskHandler {
+
+    private final OrderMadeSearchService caseSearchService;
+
+    private final CoreCaseDataService coreCaseDataService;
+
+    private final CaseDetailsConverter caseDetailsConverter;
+    private final ObjectMapper mapper;
+
+    @Override
+    public void handleTask(ExternalTask externalTask) {
+        List<CaseData> cases = getOrderMadeCasesThatAreEndingToday();
+        log.info("Job '{}' found {} case(s)", externalTask.getTopicName(), cases.size());
+
+        cases.forEach(this::fireEventForStateChange);
+    }
+
+    private List<CaseData> getOrderMadeCasesThatAreEndingToday() {
+        List<CaseDetails> orderMadeCases = caseSearchService.getGeneralApplications();
+        return orderMadeCases.stream()
+            .map(caseDetailsConverter::toCaseData)
+            .filter(caseData -> caseData.getJudicialDecisionMakeOrder().getJudgeApproveEditOptionDate() != null
+                && caseData.getGeneralAppType().getTypes().contains(STAY_THE_CLAIM)
+                && caseData.getJudicialDecisionMakeOrder().getEsOrderProcessedByStayScheduler().equals(YesOrNo.NO)
+                && (!now().isBefore(caseData.getJudicialDecisionMakeOrder().getJudgeApproveEditOptionDate()))
```

Comment:
Please revisit the line 52 for field name @Azam-Hmcts

Reply:
refactored code as per comments